bundle/edu.gemini.spdb.reports.collection/src/main/scala/edu/gemini/dbTools/timingwindowcheck/package.scala

```diff
@@ -0,0 +1,39 @@
+package edu.gemini.dbTools
+
+import java.time.Instant
+
+import scalaz._
+import Scalaz._
+import edu.gemini.pot.sp.SPObservationID
+import edu.gemini.spModel.core.SPProgramID
+import scalaz.effect.IO
+
+package object timingwindowcheck {
+
+  type Action[A] = EitherT[IO, Throwable, A]
+
+  object Action {
+
+    def unit: Action[Unit] =
+      EitherT.right(IO.ioUnit)
+
+    def fromNullableOp[A](failMessage: => String)(a: => A): Action[A] =
+      fromOption(failMessage)(Option(a))
+
+    def fromOption[A](failMessage: => String)(oa: => Option[A]): Action[A] =
+      EitherT.fromDisjunction[IO](oa \/> new RuntimeException(failMessage))
+
+    def catchLeft[A](a: => A): Action[A] =
+      EitherT(IO(a).catchLeft)
+
+    def delay[A](a: => A): Action[A] =
+      EitherT.right(IO(a))
+
+    def fail[A](failMessage: => String): Action[A] =
+      EitherT.left(IO(new RuntimeException(failMessage)))
+
+  }
+
+  type TimingWindows = Map[SPProgramID, List[(SPObservationID, Instant)]]
```

Comment:
This name is a bit misleading because we don't have timing windows here but rather the instant that the last timing window expires.  Also, do we need a `Map` here?  I don't think we ever do any lookup by program id?